doc/model.md

```diff
@@ -0,0 +1,74 @@
+# Istio Service Model
```

Comment:
Can we find a different term than "Service Model"?

To date within Google, the 'service model' refers to the service broker & service catalog stuff that is being developed under the Plori code name. Reusing this term explicitly in Istio will confuse people to no end.

Perhaps just "Service Versions"?